### PR TITLE
Updated to latest gfx and use crates.io dependencies

### PR DESCRIPTION
**Pull request by mitchmindtree**
This needs to be published before I can [add the examples back](https://github.com/PistonDevelopers/gfx_graphics/pull/245) into the gfx_graphics crate (I don't have publish auth).

Ping @bvssvni :smile_cat:

### CODE REVIEW

**Comment by kvark** [comment]
Why do you need to store a `Factory`?

File: Cargo.toml [hunk 1]
```diff
@@ -15,22 +15,14 @@ homepage = "https://github.com/pistondevelopers/piston_window"
...
 name = "piston_window"
 
-[dependencies.piston]
-#git = "https://github.com/pistondevelopers/piston"
-version = "0.1.3"
 
-[dependencies.gfx]
-git = "https://github.com/gfx-rs/gfx-rs"
+[dependencies]
+gfx = "0.6.0"
```
Comment (kvark) [inline Cargo.toml]: Should be `0.6.*` instead

File: Cargo.toml [hunk 2]
```diff
@@ -15,22 +15,14 @@ homepage = "https://github.com/pistondevelopers/piston_window"
...
+gfx_device_gl = "0.4.0"
```
Comment (kvark) [inline Cargo.toml]: Same here, please use `0.4.*`

**Comment by mitchmindtree** [comment]
Is there a better way?

The reason I'm storing it is because it is constructed via `create` along with the device to have access to it. So the situation [here] would become:
```Rust
    let rust_logo = Texture::from_path(events.factory.borrow_mut().deref_mut(),
                                       &Path::new("./assets/rust.png"),
                                       &TextureSettings::new()).unwrap();

```

What would you recommend instead? To return the factory in a tuple alongside the window in `PistonWindow::new`'s return?

**Comment by kvark** [comment]
Ah, I see, I didn't realize everything is public. Let it be there then.
Will you change the dependency versions?

**Comment by mitchmindtree** [comment]
Done :smile:

**Comment by kvark** [comment]
Thanks!

**Pull request merged**